dora/src/cannon/codegen.rs

```diff
@@ -938,6 +990,191 @@ where
             self.asm.bind_label_to(jump.label, offset);
         }
     }
+
+    fn emit_new_object(&mut self, dest: Register, class_def_id: ClassDefId) {
+        assert_eq!(self.bytecode.register_type(dest), BytecodeType::Ptr);
+
+        let cls = self.vm.class_defs.idx(class_def_id);
+        let cls = cls.read();
+
+        self.asm.emit_comment(Comment::Alloc(class_def_id));
+
+        let alloc_size = match cls.size {
+            InstanceSize::Fixed(size) => AllocationSize::Fixed(size as usize),
+            _ => unimplemented!(
+                "class size type {:?} for new object not supported",
+                cls.size
+            ),
+        };
+
+        let array_ref = match cls.size {
+            InstanceSize::ObjArray => true,
+            _ => false,
+        };
+
+        let gcpoint = GcPoint::from_offsets(self.references.clone());
```

Comment:
Right now we need to pass a GcPoint in here, with Cannon this is probably not needed anymore. We might want to change this later.